src/Configuration/Config.AzureKeyVault/src/AzureKeyVaultConfigurationProvider.cs

```diff
@@ -78,6 +108,43 @@ private async Task LoadAsync()
             } while (secrets != null);
 
             Data = data;
+            OnReload();
+
+            if (_reloadOnChange && _pollingTask == null)
+            {
+                _pollingTask = PollForSecretChangesAsync(_reloadPollDelay, _cancellationToken.Token);
+            }
+        }
+
+        private async Task<bool> ShouldReloadAsync()
+        {
+            var secrets = await _client.GetSecretsAsync(_vault).ConfigureAwait(false);
+            if (secrets.Count() != _loadedSecrets.Count())
+            {
+                return true;
+            }
+
+            foreach (var secret in secrets)
+            {
+                if (!_loadedSecrets.ContainsKey(secret.Identifier.Name))
+                {
+                    return true;
+                }
+
+                long? currentSecretLastUpdateTick = secret.Attributes.Updated.Value.Ticks;
+                long? loadedSecretLastUpdateTick = _loadedSecrets[secret.Identifier.Name].Updated.Value.Ticks;
+
+                if(currentSecretLastUpdateTick != null && loadedSecretLastUpdateTick != null)
+                {
+                    bool isKeyUpdateTimeDifferent = (loadedSecretLastUpdateTick != currentSecretLastUpdateTick);
+
+                    if (isKeyUpdateTimeDifferent)
+                    {
+                        return true;
```

Comment:
can't you just return isKeyUpdateTimeDifferent here? If its true then it will return true, otherwise it would return false anyways

Reply:
We're trying to break out of the loop and return if any of the secrets have changed.

Reply:
Ah you're right, makes sense